File: src/JavaToDoApi/src/main/java/com/fils/dal/ToDoRepository.java
package com.fils.dal;

import com.fils.models.ToDoItem;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/**
 * Created by radu on 16/11/2016.
 */
public class ToDoRepository {
    private List<ToDoItem> _toDoList;

    public ToDoRepository(){
        _toDoList = new ArrayList<>();

        _toDoList.add(new ToDoItem("Get groceries", "43.465187,-80.522372", false));
        _toDoList.add(new ToDoItem("Walk the dog", "43.465187,-80.522372", false));
        _toDoList.add(new ToDoItem("Take the car to the shop", "43.465187,-80.522372", false));
    }

    public List<ToDoItem> getToDoItems(){
        return _toDoList;
    }

    public ToDoItem getById(UUID id){
        for(ToDoItem t : _toDoList)
            if(t.getId().compareTo(id) == 0)
                return t;

        return null;
    }

    public void createToDoItem(ToDoItem toDo){
        _toDoList.add(toDo);
    }

    public void deleteToDoItem(UUID id){

        ToDoItem toDoToDelete = null;

        for(ToDoItem t : _toDoList){
            if(t.getId().compareTo(id) == 0)
                toDoToDelete = t;
        }

        if(toDoToDelete != null)
            _toDoList.remove(toDoToDelete);
    }

    public void updateToDoItem(ToDoItem toDo){

        ToDoItem toDoToUpdate = null;

        for(ToDoItem t : _toDoList){
            if(t.getId().compareTo(toDo.getId()) == 0)
                toDoToUpdate = t;
        }

        if(toDoToUpdate != null)
            toDoToUpdate.UpdateToDo(toDo);
    }
}

File: src/AspNetToDoApi/AspNetToDoApi/DAL/ToDoRepository.cs
using AspNetToDoApi.Models;
using System;
using System.Collections.Generic;
using System.Linq;

namespace AspNetToDoApi.DAL
{
    public class ToDoRepository
    {
        private List<ToDoItem> _toDoList { get; set; }

        public ToDoRepository()
        {
            _toDoList = new List<ToDoItem>()
            {
                new ToDoItem("Get groceries", "43.465187,-80.522372", false),
                new ToDoItem("Walk the dog", "43.465187,-80.522372", false),
                new ToDoItem("Take the car to the shop", "43.465187,-80.522372", false)
            };
        }

        public List<ToDoItem> GetToDoItems()
        {
            return _toDoList;
        }

        public void CreateToDo(ToDoItem item)
        {
            if (item.Id == Guid.Empty)
                item.Id = Guid.NewGuid();
            _toDoList.Add(item);
        }

        public void Delete(Guid id)
        {
            var toDo = _toDoList.FirstOrDefault(t => t.Id == id);

            if (toDo != null)
                _toDoList.Remove(toDo);
        }

        public void Update(ToDoItem item)
        {
            var toDoToUpdate = _toDoList.FirstOrDefault(t => t.Id == item.Id);

            if (toDoToUpdate != null)
                toDoToUpdate.UpdateToDo(item);
        }

        public ToDoItem GetById(Guid id)
        {
            return _toDoList.FirstOrDefault(t => t.Id == id);
        }
    }
}
